File: 0022-MajorityElement/169-MajorityElement.java
import java.util.Hashtable;

class Solution {
    public int majorityElement(int[] nums) {
        Hashtable<Integer, Integer> myMap = new Hashtable<>();
        for(int n : nums){
            myMap.put(n,myMap.getOrDefault(n, 0 ) + 1 );

        }
        int max = 0;
        int  result = nums[0] ;
        for(int i : myMap.keySet()){
            if(myMap.get(i) > max){
                max= myMap.get(i);
                result= i;
            }

        }
        return result;
    }
}

File: 0022-MajorityElement/169-Majority Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    const hash = new Map(); 
    const max= Math.ceil(nums.length / 2);
    for (i=0;i<nums.length;i++) {   
      if(hash.has(nums[i])){       
          hash.set(nums[i],hash.get(nums[i]) + 1);    
      }   
      else{
          hash.set(nums[i],1)
      }
       if(hash.get(nums[i]) +1 > max){
         
           return nums[i];
  
       }
    }
   
   
  };